sql/_32_damson/cbrd_22783_REGEXP_functions/cases/REGEXP_REPLACE.sql

```diff
@@ -0,0 +1,64 @@
+--mandatory argument
+SELECT REGEXP_REPLACE ('TechOnTheNet is a great resource', '^(\S*)', 'CheckYourMath');
+SELECT REGEXP_REPLACE();
+SELECT REGEXP_REPLACE ('abc');
+SELECT REGEXP_REPLACE ('aabbb', 'ab{-1,3}$');
+SELECT REGEXP_REPLACE ('', 'a|e|i|o|u', '#');
+SELECT REGEXP_REPLACE ('abc', '', '#');
+SELECT REGEXP_REPLACE ('abc', 'a|e|i|o|u', '');
+SELECT REGEXP_REPLACE ('', '', '#');
+SELECT REGEXP_REPLACE ('', 'a|e|i|o|u', '');
+SELECT REGEXP_REPLACE ('abc', '', '');
+SELECT REGEXP_REPLACE ('', '', '');
+SELECT REGEXP_REPLACE (NULL, 'a|e|i|o|u', '#');
+SELECT REGEXP_REPLACE ('abc', NULL, '#');
+SELECT REGEXP_REPLACE ('abc', 'a|e|i|o|u', NULL);
+SELECT REGEXP_REPLACE (NULL, NULL, '#');
+SELECT REGEXP_REPLACE (NULL, 'a|e|i|o|u', NULL);
+SELECT REGEXP_REPLACE ('abc', NULL, NULL);
+SELECT REGEXP_REPLACE (NULL, NULL, NULL);
+
+--optional arguments
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 6);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 'a');
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 0);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', -5);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 9999);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', NULL, 2);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 6);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 6, 'a');
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 2, 0);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 2, -1);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 1, 9999);
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 1, NULL); 
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 3, 2, 'i');
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 3, 2, 'c');
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 1, 0, '');
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 1, 2, 'ia');
+SELECT REGEXP_REPLACE ('TechOnTheNet', 'a|e|i|o|u', '#', 1, 2, NULL);
+
+--functional test
+SELECT 
+REGEXP_REPLACE (name, '[a-d]', '#'), REGEXP_REPLACE (name, '[e-z]', '@') 
+from athlete LIMIT 5;
```

Comment:
Athlete is tables that do not exist in the basic db, so create table and insert data must be performed.